tests/validation/cattlevalidationtest/core/test_container.py

```diff
@@ -117,30 +117,26 @@ def test_linking(client, test_name, managed_network):
     delete_all(client, [link_client, link_server, link_server2])
 
 
-def create_container(admin_client, net):
-    cmd = '/bin/bash -c "sleep 5; ip addr show eth0"'
-    c = admin_client.create_container(imageUuid=TEST_IMAGE_UUID,
-                                      networkIds=[net.id],
-                                      command=cmd)
-    c = admin_client.wait_success(c)
-    logs = c.logs()
-
-    return logs, c, c.primaryIpAddress
-
-
-def cleanup(container):
-        try:
-            container.stop(remove=True, timeout=0)
-        except:
-            # We've tried our best!
-            pass
+def test_ip_inject(client, managed_network, test_name):
+    cleanup_items = []
+    try:
```

Comment:
It is ***always*** cattle server. With native containers, the user must add an environment variable to the container `RANCHER_NETWORK=true` (will be a label in the future). When the native container's existence is reported to cattle, cattle will allocate an IP as part of the normal container create process if that env var is present.